nipype/interfaces/niftyreg/regutils.py

```diff
@@ -0,0 +1,739 @@
+    'reg_resample -flo im2.nii -inter 1 -omp 4 -ref im1.nii -trans \
+warpfield.nii -res .../im2_res.nii.gz'
+
+    """
+    _cmd = get_custom_path('reg_resample')
+    input_spec = RegResampleInputSpec
+    output_spec = RegResampleOutputSpec
+
+    # Need this overload to properly constraint the interpolation type input
+    def _format_arg(self, name, spec, value):
+        if name == 'inter_val':
+            inter_val = {'NN': 0, 'LIN': 1, 'CUB': 3, 'SINC': 5}
+            return spec.argstr % inter_val[value]
+        else:
+            return super(RegResample, self)._format_arg(name, spec, value)
+
+    def _gen_filename(self, name):
+        if name == 'out_file':
+            return self._gen_fname(self.inputs.flo_file,
+                                   suffix='_%s' % self.inputs.type,
+                                   ext='.nii.gz')
+        return None
+
+
+class RegJacobianInputSpec(CommandLineInputSpec):
+    """ Input Spec for RegJacobian. """
+    # Reference file name
+    desc = 'Reference/target file (required if specifying CPP transformations.'
+    ref_file = File(exists=True,
+                    desc=desc,
+                    argstr='-ref %s')
+    # Input transformation file
+    trans_file = File(exists=True,
+                      desc='The input non-rigid transformation',
+                      argstr='-trans %s',
+                      mandatory=True)
+    type = traits.Enum('jac', 'jacL', 'jacM',
+                       usedefault=True,
+                       argstr='-%s',
+                       position=-2,
+                       desc='Type of jacobian outcome')
+    out_file = File(genfile=True,
+                    desc='The output jacobian determinant file name',
+                    argstr='%s',
+                    position=-1)
+    # Set the number of omp thread to use
+    omp_core_val = traits.Int(desc='Number of openmp thread to use',
+                              argstr='-omp %i')
+
+
+class RegJacobianOutputSpec(TraitedSpec):
+    """ Output Spec for RegJacobian. """
+    out_file = File(desc='The output file')
+
+
+class RegJacobian(NiftyRegCommand):
+    """Interface for executable reg_resample from NiftyReg platform.
+
+    Tool to generate Jacobian determinant maps from transformation
+    parametrisation generated by reg_f3d
+
+    For source code, see https://cmiclab.cs.ucl.ac.uk/mmodat/niftyreg
+
+    Examples
+    --------
+    >>> from nipype.interfaces import niftyreg
+    >>> node = niftyreg.RegJacobian()
+    >>> node.inputs.ref_file = 'im1.nii'
+    >>> node.inputs.trans_file = 'warpfield.nii'
+    >>> node.inputs.omp_core_val = 4
+    >>> node.cmdline  # doctest: +ELLIPSIS +ALLOW_UNICODE
+    'reg_jacobian -omp 4 -ref im1.nii -trans warpfield.nii -jac \
+.../warpfield_jac.nii.gz'
+
+    """
+    _cmd = get_custom_path('reg_jacobian')
+    input_spec = RegJacobianInputSpec
+    output_spec = RegJacobianOutputSpec
+
+    def _gen_filename(self, name):
+        if name == 'out_file':
```

Comment:
How should I proceed to name the file with name_source/name_template in the case of:
the file will be named:  '{0}_{1}.ext'.format(self.inputs.trans_file, self.inputs.type)?